CI/Dockerfile

```diff
@@ -34,63 +55,139 @@ RUN apt-get -y update; \
     pip install cython;
 
 
-# Copy scripts to docker image
-RUN mkdir -p /root/etc/
-COPY CI/ /root/etc/CI
-#TODO move sh file contents into this Dockerfile
-ENV docker_env=/root/etc/CI/env.sh
-
-ENV build_dir=/root/build_dir
-ENV install_dir=/root/opt
-
-
 FROM base as external_deps
 
-#setting the COMPILER variable
-ARG COMPILER=gcc
-ENV COMPILER=${COMPILER}
-
-# Set Geant4 env variable
+ARG build_dir
+ARG install_dir
+ARG COMPILER
+ARG ci_jobs
+ARG geant4_version
+ARG CC
+ARG CXX
+ARG EMBREE_BRANCH
+
+ENV geant4_basename=geant4-v${geant4_version}
+ENV geant4_tarball=${geant4_basename}.tar.gz
+ENV geant4_shasum=2397eb859dc4de095ff66059d8bda9f060fdc42e10469dd7890946293eeb0e39
 ENV geant4_build_dir=${build_dir}/geant4
 ENV geant4_install_dir=${install_dir}/geant4
 
-# Build Geant4
-#TODO move sh file contents into this Dockerfile
-RUN /root/etc/CI/docker/build_geant4.sh
-
+RUN mkdir -p ${geant4_build_dir}/bld
+RUN cd ${geant4_build_dir} && \
+    wget https://gitlab.cern.ch/geant4/geant4/-/archive/v${geant4_version}/${geant4_tarball} --no-check-certificate && \
+    tar -xzf ${geant4_tarball} && \
+    cd bld && \
+    cmake ../${geant4_basename} -DBUILD_STATIC_LIBS=ON \
+                -DGEANT4_USE_SYSTEM_EXPAT=OFF \
+                -DCMAKE_BUILD_TYPE=Release \
+                -DCMAKE_C_COMPILER=${CC} \
+                -DCMAKE_CXX_COMPILER=${CXX} \
+                -DCMAKE_INSTALL_RPATH=${geant4_install_dir}/lib \
+                -DCMAKE_INSTALL_PREFIX=${geant4_install_dir} && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${geant4_build_dir}
+
+# these two double down ENVs don't appear to be used yet
+# However they were in orginal dockerfile, so they have been kept
+# TODO add double down to the dockerfile
 ENV double_down_build_dir=${build_dir}/double-down/
 ENV double_down_install_dir=${install_dir}/double-down/
 
-# Build Embree
-#TODO move sh file contents into this Dockerfile
-RUN /root/etc/CI/docker/build_embree.sh
+ENV embree_install_dir=${install_dir}/embree
+
+RUN mkdir EMBREE && \
```

Comment:
This would be a good opportunity to impose some consistency in how we use variables and directory structures across our different builds.

For example, we could define the following variables for each package, `pkg`:
* `pkg_build_dir`
* `pkg_install_dir`
And then always use those directories as we `mkdir`, `cd`, etc

Reply:
yep that sounds good, these sorts of improvements become more obvious when we see all the different ways the install scripts opperated in one place

Reply:
I've had a bit of a tiny up from the original PR that was just a copy paste of the install scripts. I think the build folders are now more consistent